app/gui/src/model/execution_context.rs

```diff
@@ -271,6 +273,43 @@ pub struct AttachedVisualization {
 
 
 
+// ======================
+// === ComponentGroup ===
+// ======================
+
+/// A named group of components (language elements displayed by the Component Browser).
+///
+/// Components are displayed in the Commponent Browser in groups, and such groups can be defined by
+/// libraries imported into the current execution context. To learn more about component groups,
+/// see the [Component Browser Design
+/// Document](https://github.com/enso-org/design/blob/e6cffec2dd6d16688164f04a4ef0d9dff998c3e7/epics/component-browser/design.md).
+#[allow(missing_docs)]
+#[derive(Clone, Debug, PartialEq)]
+pub struct ComponentGroup {
+    pub name:       ImString,
+    /// An optional color to use when displaying the component group.
+    pub color:      Option<color::Rgb>,
+    pub components: Vec<SuggestionQualifiedName>,
+}
+
+impl ComponentGroup {
+    /// Construct from a Language Server type representing a component group.
```

Comment:
1. Does this function consumes a Language Server and outputs a type representing component group? I don't think so. I don't see it consuming Language Server.
2. What is "a type representing component group"? The doc for `ComponentGroup` tell that this is named group of components. I mean, this is confusing - why we are using so many names for the same thing?